linkml_runtime/loaders/loader_root.py

```diff
@@ -132,53 +137,52 @@ def loads(
         return self.load(source, target_class, metadata=metadata)
 
     def _construct_target_class(
-        self, data_as_dict: Union[dict, list[dict]], target_class: Union[type[YAMLRoot], type[BaseModel]]
-    ) -> Optional[Union[BaseModel, YAMLRoot, list[BaseModel], list[YAMLRoot]]]:
+        self, data_as_dict: dict | list[dict], target_class: type[YAMLRoot | BaseModel]
+    ) -> BaseModel | YAMLRoot | list[BaseModel] | list[YAMLRoot] | None:
         if data_as_dict:
             if isinstance(data_as_dict, list):
                 if issubclass(target_class, YAMLRoot):
                     return [target_class(**as_dict(x)) for x in data_as_dict]
-                elif issubclass(target_class, BaseModel):
+                if issubclass(target_class, BaseModel):
                     return [target_class.model_validate(as_dict(x)) for x in data_as_dict]
-                else:
-                    raise ValueError(f"Cannot load list of {target_class}")
-            elif isinstance(data_as_dict, dict):
+                msg = f"Cannot load list of {target_class}"
+                raise ValueError(msg)
+            if isinstance(data_as_dict, dict):
                 if issubclass(target_class, BaseModel):
                     return target_class.model_validate(data_as_dict)
                 return target_class(**data_as_dict)
 
             if isinstance(data_as_dict, JsonObj):
                 return [target_class(**as_dict(x)) for x in data_as_dict]
-            else:
-                raise ValueError(f"Unexpected type {data_as_dict}")
-        else:
-            return None
+
+            msg = f"Unexpected type {data_as_dict}"
```

Comment:
separate error message construction

removed surplus `else` statements to simplify code